SU2_CFD/src/solvers/CNSSolver.cpp

```diff
@@ -613,6 +613,144 @@ su2double CNSSolver::GetCHTWallTemperature(const CConfig* config, unsigned short
+    const su2double Prandtl_Lam = config->GetPrandtl_Lam();
+    const su2double Prandtl_Turb = config->GetPrandtl_Turb();
+    const su2double Gas_Constant = config->GetGas_ConstantND();
+    const su2double Cp = (Gamma / Gamma_Minus_One) * Gas_Constant;
+    
+    su2double **Jacobian_i = nullptr;
+    if (implicit) {
+      Jacobian_i = new su2double* [nVar];
+      for (auto iVar = 0u; iVar < nVar; iVar++)
+        Jacobian_i[iVar] = new su2double [nVar] ();
+    }
+
+    /*--- Loop over boundary points ---*/
+
+    SU2_OMP_FOR_DYN(OMP_MIN_SIZE)
+    for (auto iVertex = 0u; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+
+      const auto iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+
+      if (!geometry->nodes->GetDomain(iPoint)) continue;
+
+      /*--- Compute dual-grid area and boundary normal ---*/
+
+      const auto Normal = geometry->vertex[val_marker][iVertex]->GetNormal();
+
+      su2double Area = GeometryToolbox::Norm(nDim, Normal);
+
+      su2double UnitNormal[MAXNDIM] = {0.0};
+      for (auto iDim = 0u; iDim < nDim; iDim++)
+        UnitNormal[iDim] = -Normal[iDim]/Area;
+
+      /*--- Compute closest normal neighbor ---*/
+
+      const auto Point_Normal = geometry->vertex[val_marker][iVertex]->GetNormal_Neighbor();
+
+      /*--- Get coordinates of i & nearest normal and compute distance ---*/
+
+      const auto Coord_i = geometry->nodes->GetCoord(iPoint);
+      const auto Coord_j = geometry->nodes->GetCoord(Point_Normal);
+
+      su2double dist_ij = GeometryToolbox::Distance(nDim, Coord_i, Coord_j);
+
+      /*--- Store the corrected velocity at the wall which will
+       be zero (v = 0), unless there is grid motion (v = u_wall)---*/
+
+      if (dynamic_grid) {
+        nodes->SetVelocity_Old(iPoint, geometry->nodes->GetGridVel(iPoint));
+      }
+      else {
+        su2double zero[MAXNDIM] = {0.0};
+        nodes->SetVelocity_Old(iPoint, zero);
+      }
+
+      for (auto iDim = 0u; iDim < nDim; iDim++)
+        LinSysRes(iPoint, iDim+1) = 0.0;
+      nodes->SetVel_ResTruncError_Zero(iPoint);
+
+      /*--- Get transport coefficients ---*/
+
+      su2double laminar_viscosity    = nodes->GetLaminarViscosity(iPoint);
+      su2double eddy_viscosity       = nodes->GetEddyViscosity(iPoint);
+      su2double thermal_conductivity = Cp * (laminar_viscosity/Prandtl_Lam + eddy_viscosity/Prandtl_Turb);
+
+      const su2double There = nodes->GetTemperature(Point_Normal);
+      su2double Twall = nodes->GetTemperature(iPoint);
+
+      /*--- Compute the normal gradient in temperature using Twall ---*/
+
+      su2double dTdn = -(There - Twall)/dist_ij;
+
+      /*--- Apply a weak boundary condition for the energy equation.
+       Compute the residual due to the prescribed heat flux. ---*/
+
+      su2double Res_Conv = 0.0;
+      su2double Res_Visc = Area * (thermal_conductivity * dTdn - emissivity*STEFAN_BOLTZMANN*pow(Twall,4));
```

Comment:
Too much copy pasting from somewhere else :smile: 

What is the theory behind this formulation?
To what extent would it be possible and physical to combine this with the other types of wall boundary?

Reply:
<img width="992" alt="Screen Shot 2022-07-11 at 12 24 46 AM" src="https://user-images.githubusercontent.com/44848904/178188385-3cb6bc92-2450-4fe3-b0a6-126318880e9f.png">

Reply:
So the theory behind this BC is that as your wall temperature increases, physically it will start to radiate according to the Stefan-Boltzmann Law ~ q_rad = emissivity*SB_constant*(T_w^4 - T_surrounding^4). T_surrounding is generally taken as 0, unless there is a more complex treatment of the shock layer radiation (which we don't have in SU2) so I've dropped it here. Then, for an insulated wall, you'll have the convective and radiative heat fluxes in equilibrium. Adding this to the viscous flux residual and driving to 0, you impose a radiative equilibrium wall temperature at the wall. Here's a comparison of the temperature you get with an adiabatic wall vs radiative equilibrium over a range of Mach. Basically, by incorporating the large radiative heat flux you get at high temperatures, the radiative wall gives you a much more accurate prediction of wall temp.

![wall_temp_comp](https://user-images.githubusercontent.com/44848904/178189895-372b5468-1f71-4c0b-9447-f2f3e94319d8.png)

Reply:
I see, thank you for the explanation.
In that case I guess it would not be appropriate to combine this with the heat flux boundary, and certainly not with isothermal... Which is sad because there is more boiler plate you'll have to add.

Reply:
Yeah, I'm wondering how sustainable the current structure of adding a new function for each BC is. @WallyMaier floated the idea of a generic viscous wall BC, where most of the code is shared but the residual/Jacobian terms are edited based on the BC type (isothermal, HF, etc.). Is this something we might use to combat code bloat?

Reply:
It seems like you are describing exactly what we have in CNSSolver :rofl:

Those pictures are looking nice btw, they would look even better here https://su2code.github.io/tutorials/home/ or here https://su2code.github.io/vandv/home/ :smile:

Reply:
^^*wink wink*

But I think its worth discussing at the dev meeting to figure out the best way forward with additional boundaries.

Reply:
@pcarruscag  Good point, you mean in that way that the isothermal wall BC calls isothermal wall generic, etc.? Yeah I think I see what you are getting at in your comment below. And yes! Will definitely add to tutorials, currently looking for a V&V case, but before all of that putting together an SU2 conference abstract 😆 

@WallyMaier Yeah good point, I'll definitely join next Tuesday, it would be great to talk through how to do this most efficiently going forward.